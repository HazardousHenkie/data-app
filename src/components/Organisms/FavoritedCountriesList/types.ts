import { ResponseError } from 'utils/request'

export interface FavoritedCountryInterface {
    name: string
}

interface FavoritedCountriesState {
    readonly error: boolean | ResponseError
    readonly loading: boolean
    readonly countries: FavoritedCountryInterface[]
}

export default FavoritedCountriesState
