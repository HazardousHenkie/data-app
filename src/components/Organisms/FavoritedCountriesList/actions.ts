import { action } from 'typesafe-actions'

import ActionTypes from './constants'

import { CountryState } from './types'

const setSelectedCountry = (country: CountryState['country']) =>
    action(ActionTypes.GET_FAVORITED_COUNTRIES, country)

export default setSelectedCountry
