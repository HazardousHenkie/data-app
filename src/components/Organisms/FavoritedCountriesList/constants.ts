enum ActionTypes {
    GET_FAVORITED_COUNTRIES = 'components/FavoritedCountriesList/GET_FAVORITED_COUNTRIES',
    GET_FAVORITED_COUNTRIES_SUCCESS = 'components/FavoritedCountriesList/GET_FAVORITED_COUNTRIES_SUCCESS',
    GET_FAVORITED_COUNTRIES_ERROR = 'components/FavoritedCountriesList/GET_FAVORITED_COUNTRIES_ERROR'
}

export default ActionTypes
