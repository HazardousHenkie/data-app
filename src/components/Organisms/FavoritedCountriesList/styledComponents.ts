import styled from 'styled-components'

const FavoritedCountriesList = styled.div`
    position: relative;
`

export default FavoritedCountriesList
