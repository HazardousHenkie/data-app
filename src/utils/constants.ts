enum ENDPOINTS {
    LOGOUT = '/.netlify/functions/googleLogout',
    lOGIN = '/.netlify/functions/googleLogin',
    REFRESH_TOKEN = '/.netlify/functions/refreshToken',
    COUNTRIES = '/.netlify/functions/countries',
    FAVORITE_COUNTRIES = '/.netlify/functions/getFavoritedCountries',
    SAVE_COUNTRY = '/.netlify/functions/saveCountry/',
    DELETE_COUNTRY = '/.netlify/functions/deleteCountry/'
}

export default ENDPOINTS
