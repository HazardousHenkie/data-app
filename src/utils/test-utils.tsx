import React from 'react'
import { render } from '@testing-library/react'
import configureStore from 'configureStore'

import history from 'utils/history'
import { Provider } from 'react-redux'

import variables from 'styles/variables'
import { ThemeProvider } from 'styled-components'
import { ThemeProvider as MuiThemeProvider } from '@material-ui/styles'
import lightTheme from 'styles/themeStyles'

const customRender = (
    ui: React.ReactElement,
    {
        initialState = {},
        store = configureStore(initialState, history),
        ...renderOptions
    } = {}
) => {
    const Wrapper: React.FC = ({ children }) => {
        return (
            <Provider store={store}>
                <ThemeProvider theme={variables}>
                    <MuiThemeProvider theme={lightTheme}>
                        <ThemeProvider theme={lightTheme}>
                            {children}
                        </ThemeProvider>
                    </MuiThemeProvider>
                </ThemeProvider>
            </Provider>
        )
    }
    return render(ui, {
        wrapper: Wrapper,
        ...renderOptions
    })
}

export * from '@testing-library/react'

export { customRender as render }
