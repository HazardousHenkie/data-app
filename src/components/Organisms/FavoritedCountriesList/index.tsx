import React from 'react'

import { useTranslation } from 'react-i18next'

import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'

import { createStructuredSelector } from 'reselect'
import { useSelector } from 'react-redux'

import InlineLoader from 'components/Atoms/InlineLoader'

import {
    makeSelectFavoritedCountries,
    makeSelectLoader
} from 'globals/favoritedCountriesList/selectors'
import FavoritedCountriesList from './styledComponents'

const stateSelector = createStructuredSelector({
    favoritedCountries: makeSelectFavoritedCountries(),
    loading: makeSelectLoader()
})

const MenuList: React.FC = () => {
    const { t } = useTranslation('menuList')
    const { favoritedCountries, loading } = useSelector(stateSelector)

    return (
        <FavoritedCountriesList data-testid="FavoritedCountriesList">
            {loading && <InlineLoader />}

            {favoritedCountries &&
                favoritedCountries[0] &&
                favoritedCountries[0].ts !== 0 && (
                    <List
                        data-testid="FavoritedCountriesList_list"
                        aria-label={t(
                            'favoritedCountryList:titleMenu',
                            'Favorite countries'
                        )}
                    >
                        <ListItem>
                            <ListItemText
                                primary={t(
                                    'menuList:favoriteCountries',
                                    'Favorite countries'
                                )}
                            />
                        </ListItem>

                        {favoritedCountries.map(favoritedCountry => (
                            <ListItem key={favoritedCountry.data.countryId}>
                                <ListItemText
                                    primary={favoritedCountry.data.countryId}
                                />
                            </ListItem>
                        ))}
                    </List>
                )}
        </FavoritedCountriesList>
    )
}

export default MenuList
