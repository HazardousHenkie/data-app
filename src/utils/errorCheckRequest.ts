import { put } from 'redux-saga/effects'
import { getRefreshTokenRequest } from 'globals/authentication/refreshToken/actions'
import { logoutRequest } from 'globals/authentication/logout/actions'

import { setError } from 'globals/globalErrors/actions'
import { ResponseError } from './request'

// test this one?
function* requestErrorCheck(error: ResponseError) {
    try {
        if (localStorage.getItem('userId')) {
            yield put(
                getRefreshTokenRequest(localStorage.getItem('userId') as string)
            )
        } else {
            throw Error('No userId in local storage.')
        }
    } catch {
        yield put(logoutRequest())
        yield put(setError(error))
    }
}
export default requestErrorCheck
