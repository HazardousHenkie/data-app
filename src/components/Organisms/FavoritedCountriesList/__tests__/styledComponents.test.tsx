import React from 'react'
import { render } from 'utils/test-utils'

import FavoritedCountriesList from '../styledComponents'

import 'jest-styled-components'

describe('<FavoritedCountriesList />', () => {
    it('should render and match the snapshot', () => {
        const { container } = render(<FavoritedCountriesList />)

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render a <DIV> tag', () => {
        const { container } = render(<FavoritedCountriesList />)

        expect(container.firstElementChild?.tagName).toEqual('DIV')
    })
})
