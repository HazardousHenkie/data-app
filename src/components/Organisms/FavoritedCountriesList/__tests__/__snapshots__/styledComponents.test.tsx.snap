// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritedCountriesList /> should render and match the snapshot 1`] = `
.c0 {
  position: relative;
}

<div
  class="c0"
/>
`;
